06-building-a-program-start-to-finish/writing-tests/Output/tests.js

```diff
@@ -0,0 +1,190 @@
+    expect(search.hasOwnProperty('falls')).toBe(true);
+    expect(search.hasOwnProperty('hole')).toBe(true);
+  });
+  it('should return false for hasOwnProperty "minimal and sucks"', () => {
+    expect(search.hasOwnProperty('minimal')).toBe(false);
+    expect(search.hasOwnProperty('sucks')).toBe(false);
+  });
+  it('should return 0 for createIndexObject.(into and of)', () => {
+    expect(search.into).toBe[0];
+    expect(search.of).toBe[0];
+  });
+  it('should return 1 for createIndexObject.(lord and powerful)', () => {
+    expect(search.lord).toBe[1];
+    expect(search.powerful).toBe[1];
+  });
+  it('should return undefined for createIndexObject.(darklord and nothing)', () => {
+    expect(search.darklord).toBe(undefined);
+    expect(search.nothing).toBe(undefined);
+  });
+});
+
+describe('getIndex', () => {
+  const get = index.getIndex(books);
+  it('should return true for object type ', () => {
+    expect(typeof get).toBe('object');
+  });
+  it('should not return true for integer type ', () => {
+    expect(typeof get).not.toBe('integer');
+  });
+  it('should return true for hasOwnProperty "alice and wonderland"', () => {
+    expect(get.hasOwnProperty('alice')).toBe(true);
+    expect(get.hasOwnProperty('wonderland')).toBe(true);
+  });
+  it('should return false for hasOwnProperty "eiie and justapose"', () => {
+    expect(get.hasOwnProperty('eiie')).toBe(false);
+    expect(get.hasOwnProperty('justapose')).toBe(false);
+  });
+  it('should return 0 for createIndexObject.(alice and rabbit)', () => {
+    expect(get.alice).toBe[0];
+    expect(get.rabbit).toBe[0];
+  });
+  it('should return 1 for createIndexObject.(alliance and hobbit)', () => {
+    expect(get.alliance).toBe[1];
+    expect(get.hobbit).toBe[1];
+  });
+  it('should return undefined for createIndexObject.(alanta and honda)', () => {
+    expect(get.alanta).toBe(undefined);
+    expect(get.Honda).toBe(undefined);
+  });
+});
+
+describe('searchIndex', () => {
+  index.getIndex(books);
+  index.getIndex(gotBooks);
+  const search = index.searchIndex(books, 'alice');
+  const search2 = index.searchIndex(books, 'anja');
+  const search3 = index.searchIndex('all', 'hole');
+  const search4 = index.searchIndex('all', 'debby');
+  it('should expect typeof to be array', () => {
+    expect(Array.isArray(search)).toBe(true);
+  });
+  it('should expect typeof to be array', () => {
+    expect(Array.isArray(search3)).toBe(true);
+  });
+  it('should expect "alice" to be in books', () => {
+    expect(search[0].alice).toBe[1];
+  });
+  it('should expect "anja" not to be in books ', () => {
+    expect(search2[0].anja).toBe(undefined);
+  });
+  it('should expect "search for hole in all" to return [0]', () => {
+    console.log('here', search3[0]['books.json']);
+    expect(search3[0]['books.json']).toBe[{ hole: [0] }];
+  });
+  it('should expect "search for debby in all" to return undefined', () => {
+    expect(search4.debby).toBe(undefined);
+  });
+});
```

Comment:
"Ubiquitous Language," a body of knowledge that can be understood by both the developer and the customer. This ubiquitous language is used to shape and develop the requirements and testing needed, at the level of the customer's understanding.

```
describe("Calculator: add", () => {

    it("should be able to add 2 numbers together", () => {
        var total = add(1, 2);
        expect(total).toBe(3);
    });

    it("should be able to add 3 numbers together", () => {
        var total = add(1, 2, 3);
        expect(total).toBe(6);
    });
});
```
Please review your test paying more attention to the language used on your "it" and 'describe' functions